<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lifan Liang" />

<meta name="date" content="2025-01-06" />

<title>dynamic_eqtl</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">eQTL_methods</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">dynamic_eqtl</h1>
<h4 class="author">Lifan Liang</h4>
<h4 class="date">2025-01-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>eQTL_methods/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstreeda0ab725ca23b6fb09c76f07ea2a2d6fc564f435targetblankda0ab72a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LifanLiang/eQTL_methods/tree/da0ab725ca23b6fb09c76f07ea2a2d6fc564f435" target="_blank">da0ab72</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstreeda0ab725ca23b6fb09c76f07ea2a2d6fc564f435targetblankda0ab72a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LifanLiang/eQTL_methods/tree/da0ab725ca23b6fb09c76f07ea2a2d6fc564f435" target="_blank">da0ab72</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

Untracked files:
    Untracked:  test_matrixeqtl.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/dynamic_eqtl.Rmd</code>) and HTML
(<code>docs/dynamic_eqtl.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/da0ab725ca23b6fb09c76f07ea2a2d6fc564f435/analysis/dynamic_eqtl.Rmd" target="_blank">da0ab72</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/6bfdbb43e58918b8e893e5e7f2b5c2d9ac3f01f3/docs/dynamic_eqtl.html" target="_blank">6bfdbb4</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f89e61b7f3322c6017b0959485f39627f1034656/analysis/dynamic_eqtl.Rmd" target="_blank">f89e61b</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/629cc3630bb3566c745104de38ee2093ac26fd28/docs/dynamic_eqtl.html" target="_blank">629cc36</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/704016c709a2794fdd7b0b72494f451c7f14a217/analysis/dynamic_eqtl.Rmd" target="_blank">704016c</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/fb9f866236c21db5e99d9367066f02fd0d40b647/docs/dynamic_eqtl.html" target="_blank">fb9f866</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/c1a2fa73c15fc9ec8223d438fdd2375c65375f6b/analysis/dynamic_eqtl.Rmd" target="_blank">c1a2fa7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="model-formulation" class="section level2">
<h2>Model formulation</h2>
<p>Suppose we have expression data of <span
class="math inline">\(n\)</span> samples from <span
class="math inline">\(D\)</span> donors. Here a sample could be a single
cell, or a meta-cell, or some pseudobulk sample (of many cells) with the
same type/state. For each sample, we denote <span
class="math inline">\(X_i\)</span> as the context of interest,
e.g. treatment, cell state (such as differentiation pseudotime).
Additionally, genotype <span class="math inline">\(G_d, 1 \leq d \leq
D\)</span> may also affect expression. There is also a possible
interaction effect between genotype <span
class="math inline">\(G_d\)</span> and the context <span
class="math inline">\(X_i\)</span>. Our model of gene expression is:
<span class="math display">\[\begin{equation}
    y_i = \mu + \beta_X X_i + \beta_G G_d + \gamma X_i \cdot G_d + u_d +
\epsilon_i,
\end{equation}\]</span> where <span
class="math inline">\(\beta_X\)</span> and <span
class="math inline">\(\beta_G\)</span> are context, and genotype
effects, respectively, <span class="math inline">\(\gamma\)</span> is
the interaction effect, and <span class="math inline">\(u_d\)</span> is
the donor-specific random effect. For simplicity, we do not include some
covariates such as sex and age of donors, and percent of mitochondrial
reads of cells.</p>
<p>Fitting the model is computationally expensive. We can rewrite the
model to remove the donor random effect. We remove from <span
class="math inline">\(y_i\)</span>, the average expression of all
samples from the donor <span class="math inline">\(d\)</span>, <span
class="math inline">\(\bar{y}_d\)</span>:<br />
<span class="math display">\[\begin{equation}
    \Delta_i = y_i - \bar{y}_d.
\end{equation}\]</span> Now we can write our model as: <span
class="math display">\[\begin{equation}
    \Delta_i = \beta_X X_i + \gamma X_i G_d + \epsilon_i = X_i (\beta_X
+ \gamma G_d) + \epsilon_i.
\end{equation}\]</span> From this model, we can see that the effect of
<span class="math inline">\(X_i\)</span> on gene expression depends on
the genotype <span class="math inline">\(G_d\)</span>, denoted as <span
class="math display">\[\begin{equation}
    \theta_d = \beta_X + \gamma G_d
\end{equation}\]</span> This can be viewed as:
<code>genes modify the environmental effect</code>.</p>
</div>
<div id="model-fitting" class="section level2">
<h2>Model fitting</h2>
<p>The simplest way to solve the model is a two-step linear regression.
We first regress environment <span class="math inline">\(X\)</span>
against <span class="math inline">\(\Delta_i\)</span> to estimate the
total effect for each donor, <span
class="math inline">\(\theta_d\)</span>. And the corresponding standard
error <span class="math inline">\(\sigma_d\)</span>.</p>
<p><span class="math display">\[
\hat{\theta_d} = (X_d^TX_d)^{-1}X_d^T \Delta_d
\]</span></p>
<p><span class="math display">\[
\sigma_d^2 = var(\hat{\theta_d}) = \dfrac{var(Y_d)}{X_d^T X_d}
\]</span></p>
<p>where <span class="math inline">\(X_d\)</span> is the environment
variable in the treatment effect.</p>
<p>Then we find the MLE for :</p>
<p><span class="math display">\[
\hat{\theta}_d \sim Normal(\beta_X + \gamma G_d, \sigma_d^2)
\]</span></p>
<p>Let us denote <span class="math inline">\(1/\sigma_d^2\)</span> as
<span class="math inline">\(W_d\)</span>, the objective function to
optimize is:</p>
<p><span class="math display">\[
\sum_d^D{W_d(\hat{\theta}_d - \beta_X - \gamma G)^2}
\]</span></p>
<p>If we centered the mean of <span
class="math inline">\(\hat{\theta}_d\)</span> (weighted by <span
class="math inline">\(W\)</span>) and suppose <span
class="math inline">\(G\)</span> is standardized (mean zero in single
cell level, equivalent to <span class="math inline">\(\bar G=0\)</span>
in the next section), then <span
class="math inline">\(\beta_X=0\)</span>. The estimate of <span
class="math inline">\(\gamma\)</span> is:</p>
<p><span class="math display">\[
\hat{\gamma} = \dfrac{\sum W_d \theta&#39;_d G_d}{\sum W_d G_d^2}
\\
\theta&#39;_d = \hat \theta_d - \bar \theta
\\
\bar \theta = \dfrac{\sum W_d \hat{\theta_d}}{\sum W_d}
\]</span></p>
<p>As for the standard error of <span class="math inline">\(\hat
\gamma\)</span>, we have</p>
<p><span class="math display">\[
var(\hat \gamma) = \dfrac{1}{\sum W_dG_d^2} = \dfrac{var(Y)}{\sum G_i^2
X_i^2}
\]</span></p>
</div>
<div id="equivalence-to-the-cell-level-interaction-model"
class="section level2">
<h2>Equivalence to the cell level interaction model</h2>
<p>Alternative to the two-step model above, we can directly fit an
interaction model on single-cell level:</p>
<p><span class="math display">\[
\Delta_i = \beta_x X + \gamma X\cdot G
\]</span></p>
<p>First of all, the weighted average of <span
class="math inline">\(\hat{\theta_d}\)</span> is identical to MLE of
<span class="math inline">\(\beta_x\)</span> given that genotype has
mean zero. Intuitively, this is correct since</p>
<p><span class="math display">\[
E(\theta) = E(\beta_x+\gamma G) = \beta_x + \gamma E(G)
\]</span></p>
<p>More formally, we first derive estimator of <span
class="math inline">\(\bar \theta\)</span> and <span
class="math inline">\(\bar G\)</span> on cell level. Based on inverse
variance weighted (IVW) meta-analysis or direct derivation, the weighted
average of <span class="math inline">\(\hat{\theta}\)</span> is</p>
<p><span class="math display">\[
\bar \theta = \dfrac{\sum W_d \theta_d}{\sum W_d} = \dfrac{X^T
\Delta}{X^T X}
\]</span></p>
<p>Similarly, the weighted average of <span
class="math inline">\(G\)</span> is</p>
<p><span class="math display">\[
\bar G = \dfrac{\sum W_d G}{\sum W_d} = \dfrac{X^T G&#39; X}{X^T X} =
\dfrac{\sum G_i X_i^2}{\sum X_i^2}
\]</span></p>
<p>where <span class="math inline">\(G&#39;\)</span> is a <span
class="math inline">\(n\times n\)</span> diagonal matrix with the
diagonal elements as the genotypes</p>
<p>Since the log Likelihood for the interaction model is:</p>
<p><span class="math display">\[
\mathcal{L}(\Delta;\beta_x,\gamma) \propto
-\sum(\Delta_i-\beta_xX_i-\gamma G_i\cdot X_i)^2
\]</span></p>
<p>By setting the <span
class="math inline">\(\dfrac{d\mathcal{L}(\Delta;\beta_x,\gamma)}{d\beta_x}=0\)</span>
to 0, we should find the MLE for <span
class="math inline">\(\beta_x\)</span> as:</p>
<p><span class="math display">\[
\hat \beta_x = \dfrac{\sum\Delta_iX_i}{\sum X_i^2} - \gamma \dfrac{\sum
G_i X_i^2}{\sum X_i^2} = \bar \theta - \gamma \bar G
\]</span></p>
<p>Therefore, by setting <span class="math inline">\(\bar G =
0\)</span>, the estimator for <span class="math inline">\(\hat
\beta_x\)</span> is identical to that of <span
class="math inline">\(\bar \theta\)</span>.</p>
<p>Similarly, we can derive the MLE of <span
class="math inline">\(\gamma\)</span>. Let us denote the interaction
term <span class="math inline">\(X\cdot G\)</span> as <span
class="math inline">\(X&#39;\)</span> for convenience.</p>
<p><span class="math display">\[
\hat \gamma = \dfrac{X&#39;^T \Delta}{X&#39;^TX&#39;} - \hat \beta
\dfrac{X^TX&#39;}{X&#39;^TX&#39;}
\]</span></p>
<p>Recall that the estimate for <span
class="math inline">\(\gamma\)</span> in the two step model is:</p>
<p><span class="math display">\[
\hat \gamma = \dfrac{\sum W_d (\hat \theta_d-\bar \theta) G}{\sum W_d
G^2} = \dfrac{X&#39;^T \Delta}{X&#39;^TX&#39;} - \hat \beta
\dfrac{X^TX&#39;}{X&#39;^TX&#39;}
\]</span></p>
<p>Therefore, <span class="math inline">\(\hat \gamma\)</span> is
identical in the two approaches.</p>
<p>As for the standard error of <span class="math inline">\(\hat
\gamma\)</span>, it is obvious that they are the same by plugging in the
estimate of <span class="math inline">\(W_d\)</span> in previous
equation for <span class="math inline">\(var(\hat \gamma)\)</span>.</p>
<p>The advantage of using a two-step model instead of the interaction
model is that we decoupled the genetic variation and phenotypic
variation with an intermediate “trait” that captures the sufficient
statistics for each donor. Hence, when we need to run QTL mapping for
the same molecular trait, we only need to repeat the second step on bulk
level while retaining the power of single-cell level model. This would
bring enormous speed up to the dynamic QTL mapping procedure.</p>
</div>
<div id="simulation" class="section level2">
<h2>Simulation</h2>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
