<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lifan Liang" />

<meta name="date" content="2025-01-14" />

<title>Improving power of single cell eQTL with factorization</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">eQTL_methods</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Improving power of single cell eQTL with
factorization</h1>
<h4 class="author">Lifan Liang</h4>
<h4 class="date">2025-01-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>eQTL_methods/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250106code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250106)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250106code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250106)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstreeb196d33a49bff2c830bfea0c00728f09b4820b62targetblankb196d33a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LifanLiang/eQTL_methods/tree/b196d33a49bff2c830bfea0c00728f09b4820b62" target="_blank">b196d33</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstreeb196d33a49bff2c830bfea0c00728f09b4820b62targetblankb196d33a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LifanLiang/eQTL_methods/tree/b196d33a49bff2c830bfea0c00728f09b4820b62" target="_blank">b196d33</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory

Untracked files:
    Untracked:  data/neuron_stim_genotype_chr9_2e6-3e6.rds
    Untracked:  data/susie_code.R
    Untracked:  data/susie_simulation.zip
    Untracked:  test_matrixeqtl.R

Unstaged changes:
    Modified:   data/mixedmodel_res_phi0.02.csv
    Modified:   data/mixedmodel_res_phi0.03.csv
    Modified:   data/mixedmodel_res_phi0.04.csv
    Modified:   data/mixedmodel_res_phi0.05.csv
    Modified:   data/scalability_snp.csv
    Modified:   data/twostep_res_phi0.02.csv
    Modified:   data/twostep_res_phi0.03.csv
    Modified:   data/twostep_res_phi0.04.csv
    Modified:   data/twostep_res_phi0.05.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/factor_eqtl.Rmd</code>) and HTML
(<code>docs/factor_eqtl.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/b196d33a49bff2c830bfea0c00728f09b4820b62/analysis/factor_eqtl.Rmd" target="_blank">b196d33</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-05-12
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/1378d469af727e848d55706dfec63af87e2b6e1a/docs/factor_eqtl.html" target="_blank">1378d46</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/2745cfb061019606542cdbb1d8c68607d6a8452b/analysis/factor_eqtl.Rmd" target="_blank">2745cfb</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-11
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/2745cfb061019606542cdbb1d8c68607d6a8452b/docs/factor_eqtl.html" target="_blank">2745cfb</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-11
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/64548588aa52780211d4f0cb4da2e69f48278cf9/docs/factor_eqtl.html" target="_blank">6454858</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/a6f5a6eaaa03120e60f3954a4026aaa8fca47a55/analysis/factor_eqtl.Rmd" target="_blank">a6f5a6e</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-07
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/a6f5a6eaaa03120e60f3954a4026aaa8fca47a55/docs/factor_eqtl.html" target="_blank">a6f5a6e</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-07
</td>
<td>
wflow_publish(c("analysis", "docs"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/8e34ef652943cc26d3fb8986a76a3329666674f4/docs/factor_eqtl.html" target="_blank">8e34ef6</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/1fdfaa62a256c37750a15ddc2023a682f2159008/analysis/factor_eqtl.Rmd" target="_blank">1fdfaa6</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-06
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/factor_eqtl.html" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/0ad82da6fef4ed7d787e616ebee1bef15249abfb/analysis/factor_eqtl.Rmd" target="_blank">0ad82da</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
<td>
workflowr::wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/ca49fde79ffca8148a0210108863787e554bb231/docs/factor_eqtl.html" target="_blank">ca49fde</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/efc7cd75c9fce0bb6a77600b3ca3b3be39a9bbc3/analysis/factor_eqtl.Rmd" target="_blank">efc7cd7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-04
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/a9aaac7de843d66fdf7555bbf5eb74c2f0803465/docs/factor_eqtl.html" target="_blank">a9aaac7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/c2a480e01002956f815daf729cfcd6ad37b993f7/analysis/factor_eqtl.Rmd" target="_blank">c2a480e</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-02
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/4adac67a6015631885b579643a84a8809c67ced3/docs/factor_eqtl.html" target="_blank">4adac67</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/61fbc6c2b420cdf6f6c6e2526521de6dc1cdd5b2/analysis/factor_eqtl.Rmd" target="_blank">61fbc6c</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-30
</td>
<td>
wflow_publish(c("analysis"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/62715308c266f61cd5a6ee76996f490546b96bfe/docs/factor_eqtl.html" target="_blank">6271530</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/4b6acb635671d0a5bb484bbce346ff0b720996cc/analysis/factor_eqtl.Rmd" target="_blank">4b6acb6</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-30
</td>
<td>
wflow_publish(c("analysis"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/9a16b08556e933d1463a958dd0f54463829200fe/docs/factor_eqtl.html" target="_blank">9a16b08</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/da0ab725ca23b6fb09c76f07ea2a2d6fc564f435/analysis/factor_eqtl.Rmd" target="_blank">da0ab72</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/629cc3630bb3566c745104de38ee2093ac26fd28/docs/factor_eqtl.html" target="_blank">629cc36</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/704016c709a2794fdd7b0b72494f451c7f14a217/analysis/factor_eqtl.Rmd" target="_blank">704016c</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-23
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/651e5fcfb097d1b5599761ffb68df08f8821c0fc/docs/factor_eqtl.html" target="_blank">651e5fc</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/da3565b403066a24ee1e5719785b82e438fb81eb/analysis/factor_eqtl.Rmd" target="_blank">da3565b</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/61787887579fce98e65d82d32486249f155caabb/docs/factor_eqtl.html" target="_blank">6178788</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/0a45aff5b7a921e2fe35b424595182bbe9963811/analysis/factor_eqtl.Rmd" target="_blank">0a45aff</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/7d5eb2df353a14eb06987c99bebb5cfbe8f20919/docs/factor_eqtl.html" target="_blank">7d5eb2d</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/46baaaf4e2dbbf694005182639d5651729c700ff/analysis/factor_eqtl.Rmd" target="_blank">46baaaf</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/f0315a3632545d67c16200534b8ba125fa6f6af8/docs/factor_eqtl.html" target="_blank">f0315a3</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/a21de7d0c1805e7253125ee9941953dc94ebbd5e/analysis/factor_eqtl.Rmd" target="_blank">a21de7d</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/3512ba1a58ed454f753539c3c31a4662f27231dc/docs/factor_eqtl.html" target="_blank">3512ba1</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/6084625f84601d096b813767790d3f5f1cd5e87a/analysis/factor_eqtl.Rmd" target="_blank">6084625</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/8257db5595e539facebcdd09f0a5727a199b5e48/docs/factor_eqtl.html" target="_blank">8257db5</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/c84f46013e24ce2a1213936897a95bf05c4894eb/analysis/factor_eqtl.Rmd" target="_blank">c84f460</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/544390631ca67690af62079c7bea8123f33dfdeb/docs/factor_eqtl.html" target="_blank">5443906</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f93eb4230e56f0511cecd6d6e2c46adf568de59b/analysis/factor_eqtl.Rmd" target="_blank">f93eb42</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/fb9f866236c21db5e99d9367066f02fd0d40b647/docs/factor_eqtl.html" target="_blank">fb9f866</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/c1a2fa73c15fc9ec8223d438fdd2375c65375f6b/analysis/factor_eqtl.Rmd" target="_blank">c1a2fa7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="simulation-with-a-known-latent-factor" class="section level2">
<h2>Simulation with a known latent factor</h2>
<pre class="r"><code>sim_factor &lt;- function(alpha=0.25, beta=0.4, eps0=1.0, eps1=0.3){
  res &lt;- numeric(6L)
  names(res) &lt;- c(&quot;beta.Z&quot;,&quot;score.Z&quot;,&quot;p.Z&quot;,
                  &quot;beta.y&quot;,&quot;score.y&quot;,&quot;p.y&quot;)
  x = rnorm(100,sd=1)
  Z = alpha*x + rnorm(100,sd=eps0)
  y = beta*Z + rnorm(100,sd=eps1)
  
  res[c(&quot;beta.Z&quot;,&quot;score.Z&quot;,&quot;p.Z&quot;)] &lt;- summary(lm(Z~x))$coefficients[2,c(1,3,4)]
  res[c(&quot;beta.y&quot;,&quot;score.y&quot;,&quot;p.y&quot;)] &lt;- summary(lm(y~x))$coefficients[2,c(1,3,4)]
  
  res
}</code></pre>
<p>We generated data with one latent factor <span
class="math inline">\(Z\)</span> that affects one trait, <span
class="math inline">\(Y\)</span>. The generative process is as
follow:</p>
<p><span class="math display">\[
X \sim Normal(0,1)
\\
\epsilon_0 \sim Normal(0,\sigma_0^2)
\\
\\
\epsilon_1 \sim Normal(0,\sigma_1^2)
\\
Z = \alpha X + \epsilon_0
\\
Y = \beta Z + \epsilon_1
\]</span></p>
<p>Most parameters are fixed during simulation. We set <span
class="math inline">\(\alpha = 0.25\)</span>, <span
class="math inline">\(\sigma_0=1\)</span>, In this way, we have: <span
class="math display">\[
PVE(Z \sim X) =\dfrac{\alpha^2 var(X)}{\alpha^2 var(X) + \sigma_0^2}
\approx 5.88\%
\]</span> we set <span class="math inline">\(\beta_1 = 0.4\)</span>.
Given that :</p>
<p><span class="math display">\[
PVE(Y\sim
X)=\dfrac{\alpha^2\beta^2var(X)}{\alpha^2\beta^2var(X)+\beta^2\sigma_0^2+\sigma_1^2}
\]</span>.</p>
<p>In order to approximate realistic PVE of eQTL, which ranges around 2%
~ 5%, we set <span class="math inline">\(\sigma_1 \in
{0.2,0.3,0.4}\)</span> so that <span class="math inline">\(PVE(Y_1 \sim
X)\)</span> ranges around 3.03% and 4.76%. We compared the t score and P
values of regressing X against Y versus X against X. 100 samples were
drawn from the simulation each time.</p>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/9a16b08556e933d1463a958dd0f54463829200fe/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">9a16b08</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/651e5fcfb097d1b5599761ffb68df08f8821c0fc/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">651e5fc</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/61787887579fce98e65d82d32486249f155caabb/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">6178788</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f0315a3632545d67c16200534b8ba125fa6f6af8/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">f0315a3</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/8257db5595e539facebcdd09f0a5727a199b5e48/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">8257db5</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/fb9f866236c21db5e99d9367066f02fd0d40b647/docs/figure/factor_eqtl.Rmd/unnamed-chunk-2-1.png" target="_blank">fb9f866</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="theoretical-power-gain" class="section level3">
<h3>Theoretical power gain</h3>
<p>In fact, we can directly derive the power gain from this simple
simulation scenario. For instance, let’s focus on <span
class="math inline">\(Y\)</span>, we have:</p>
<p><span class="math display">\[
Y = \alpha \beta X + \epsilon_1 + \beta_1 \epsilon_0
\]</span></p>
<p>Hence, we can derive the expected t score of X:</p>
<p><span class="math display">\[
Tscore(Y \sim X) ^2= \dfrac{N\alpha^2
\beta^2}{\sigma_1^2+\beta^2\sigma_0^2}
\]</span> On the other hand, we have: <span class="math display">\[
Tscore(Z \sim X)^2 = \dfrac{N\alpha^2}{\sigma_0^2}
\]</span> So the expected ratio between the two should be: <span
class="math display">\[
(\dfrac{Tscore(Z \sim X)}{Tscore(Y \sim X)})^2 = 1 +
(\dfrac{\sigma_1}{\sigma_0 \beta})^2
\]</span></p>
<p>Therefore, we will always gain power if we know the latent factor as
ground truth.</p>
</div>
</div>
<div id="simulation-with-one-unknown-common-factor"
class="section level2">
<h2>Simulation with one unknown common factor</h2>
<p>This simulation follows the same scenario as the one above. The
difference is that <span class="math inline">\(Z\)</span> is unknown and
there are ten traits (<span
class="math inline">\([Y_1,Y_2,...,Y_{10}\)</span>). All <span
class="math inline">\(\beta\)</span> from <span
class="math inline">\(Z\)</span> to <span
class="math inline">\(Y\)</span> is 0.4. We performed PCA to estimate
<span class="math inline">\(Z\)</span>.</p>
<pre class="r"><code>sim_10trait &lt;- function(alpha=0.25, beta=0.4, eps0=1.0, eps1=0.3, N=100){
  x = rnorm(N,sd=1)
  Z = alpha*x + rnorm(N,sd=eps0)
  y = matrix(0,nrow = N,ncol=10)
  for(i in 1: ncol(y)){
    y[,i] &lt;- beta*Z + rnorm(N,sd=eps1)
  }
  list(x=x,Z=Z,y=y)
}</code></pre>
<div id="estimating-z" class="section level3">
<h3>Estimating Z</h3>
<p>We used the first principle component as the estimator for Z after
standardizing Y. Below is the scatter plot between <span
class="math inline">\(\hat{Z}\)</span> and <span
class="math inline">\(Z\)</span>.</p>
<pre class="r"><code>set.seed(12)
dat &lt;- sim_10trait()
y1 &lt;- scale(dat$y)
pcs &lt;- prcomp(y1)
z_hat &lt;- pcs$x[,1]
plot(z_hat, dat$Z, ylab=&quot;Oracle Z&quot;, xlab=&quot; Estimated Z&quot;)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/651e5fcfb097d1b5599761ffb68df08f8821c0fc/docs/figure/factor_eqtl.Rmd/unnamed-chunk-4-1.png" target="_blank">651e5fc</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/61787887579fce98e65d82d32486249f155caabb/docs/figure/factor_eqtl.Rmd/unnamed-chunk-4-1.png" target="_blank">6178788</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Elbow plot seems to suggest one or two PC is sufficient?</p>
<pre class="r"><code>plot(pcs$sdev^2/sum(pcs$sdev^2),ylab=&quot;PVE&quot;)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/651e5fcfb097d1b5599761ffb68df08f8821c0fc/docs/figure/factor_eqtl.Rmd/unnamed-chunk-5-1.png" target="_blank">651e5fc</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/61787887579fce98e65d82d32486249f155caabb/docs/figure/factor_eqtl.Rmd/unnamed-chunk-5-1.png" target="_blank">6178788</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="power-comparison" class="section level3">
<h3>Power comparison</h3>
<p>Repeating the simulations 300 times. Oracle Z has the highest power,
followed by <span class="math inline">\(\hat{Z} \sim x\)</span>.</p>
<pre class="r"><code>Nrep = 300

factor.res &lt;- matrix(nrow=Nrep,ncol=3)
oracle.res &lt;- matrix(nrow=Nrep,ncol=3)
marg &lt;- list()
marg$beta &lt;- matrix(nrow=Nrep,ncol=10)
marg$tstat &lt;- matrix(nrow=Nrep,ncol=10)
marg$pval &lt;- matrix(nrow=Nrep,ncol=10)
  
for(i in 1:Nrep) {
  dat &lt;- sim_10trait(N=100)
  pcs &lt;- prcomp(dat$y,center = T,scale. = T)
  z_hat &lt;- pcs$x[,1]
  factor.res[i,] &lt;- summary(lm(z_hat~dat$x))$coefficients[2,c(1,3,4)]
  oracle.res[i,] &lt;- summary(lm(dat$Z~dat$x))$coefficients[2,c(1,3,4)]
  #colnames(marg.res) &lt;- c(&quot;beta&quot;,&quot;tstat&quot;,&quot;p&quot;)
  #print(i)
  for(j in 1:ncol(dat$y)) {
    marg$beta[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,1]
    marg$tstat[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,3]
    marg$pval[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,4]
  }
}
dat1 &lt;- abs(cbind(oracle.res[,2],cbind(factor.res[,2],marg$tstat))) # Combine T statistics
colnames(dat1) &lt;- c(&quot;Oracle Z&quot;,&quot;Z hat&quot;,paste0(&quot;trait &quot;,1:10))
boxplot(dat1,las=2,ylab=&quot;t statistics&quot;)
abline(h=median(abs(factor.res[,2])),col=2,lty=2,lwd=3)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/651e5fcfb097d1b5599761ffb68df08f8821c0fc/docs/figure/factor_eqtl.Rmd/unnamed-chunk-6-1.png" target="_blank">651e5fc</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/61787887579fce98e65d82d32486249f155caabb/docs/figure/factor_eqtl.Rmd/unnamed-chunk-6-1.png" target="_blank">6178788</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The barplot below shows the frequency for P values &lt; 0.05.</p>
<pre class="r"><code>dat2 &lt;- cbind(oracle.res[,3],cbind(factor.res[,3],marg$pval)) #combine p values
colnames(dat2) &lt;- c(&quot;Oracle Z&quot;,&quot;Z hat&quot;,paste0(&quot;trait &quot;,1:10))
barplot(colMeans(dat2&lt;0.05),las=2)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/9a16b08556e933d1463a958dd0f54463829200fe/docs/figure/factor_eqtl.Rmd/unnamed-chunk-7-1.png" target="_blank">9a16b08</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="frequency-of-rejecting-null-p0.05-while-there-is-no-genetic-effect."
class="section level3">
<h3>Frequency of rejecting null (P&lt;0.05) while there is no genetic
effect.</h3>
<p>Situations where the 10 traits are still largely shared but genotype
did not affect the trait <span
class="math inline">\(\alpha=0\)</span>.</p>
<pre class="r"><code>Nrep = 300

factor.res &lt;- matrix(nrow=Nrep,ncol=3)
oracle.res &lt;- matrix(nrow=Nrep,ncol=3)
marg &lt;- list()
marg$beta &lt;- matrix(nrow=Nrep,ncol=10)
marg$tstat &lt;- matrix(nrow=Nrep,ncol=10)
marg$pval &lt;- matrix(nrow=Nrep,ncol=10)
  
for(i in 1:Nrep) {
  dat &lt;- sim_10trait(N=100, alpha=0)
  pcs &lt;- prcomp(dat$y,center = T,scale. = T)
  z_hat &lt;- pcs$x[,1]
  factor.res[i,] &lt;- summary(lm(z_hat~dat$x))$coefficients[2,c(1,3,4)]
  oracle.res[i,] &lt;- summary(lm(dat$Z~dat$x))$coefficients[2,c(1,3,4)]
  #colnames(marg.res) &lt;- c(&quot;beta&quot;,&quot;tstat&quot;,&quot;p&quot;)
  #print(i)
  for(j in 1:ncol(dat$y)) {
    marg$beta[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,1]
    marg$tstat[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,3]
    marg$pval[i,j] &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,4]
  }
}
dat3 &lt;- abs(cbind(oracle.res[,3],cbind(factor.res[,3],marg$pval))) #combine p values
colnames(dat3) &lt;- c(&quot;Oracle Z&quot;,&quot;Z hat&quot;,paste0(&quot;trait &quot;,1:10))
barplot(colMeans(dat3&lt;0.05),las=2)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/9a16b08556e933d1463a958dd0f54463829200fe/docs/figure/factor_eqtl.Rmd/unnamed-chunk-8-1.png" target="_blank">9a16b08</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-01-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="simulation-with-one-snp-affecting-one-latent-factor-and-other-snps-with-direct-effects"
class="section level3">
<h3>Simulation with one SNP affecting one latent factor and other SNPs
with direct effects</h3>
<pre class="r"><code>library(flashier)</code></pre>
<pre><code>Loading required package: ebnm</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre class="r"><code>sim_5trait &lt;- function(alpha=0.25, beta=0.4, gamma=0.05, eps0=1.0, eps1=0.3, N=100){
  x1 = rnorm(N,sd=1)
  x2 = matrix(rnorm(N*5,sd=1), nrow=N, ncol=5)
  Z = alpha*x1 + rnorm(N,sd=eps0)
  y = matrix(0,nrow = N,ncol=5)
  for(i in 1: ncol(y)){
    y[,i] &lt;- beta*Z + rnorm(N,sd=eps1) + x2[,i]*gamma
  }
  list(x1=x1,x2=x2,Z=Z,y=y)
}</code></pre>
<pre class="r"><code>dat.sim &lt;- sim_5trait()
for(i in 1:ncol(dat.sim$y)) {
  
}
summary(lm(dat.sim$y[,2]~dat.sim$x1))</code></pre>
<pre><code>
Call:
lm(formula = dat.sim$y[, 2] ~ dat.sim$x1)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.93902 -0.32141 -0.00424  0.32331  0.83516 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept) -0.02657    0.04413  -0.602    0.549
dat.sim$x1   0.05326    0.03947   1.349    0.180

Residual standard error: 0.4404 on 98 degrees of freedom
Multiple R-squared:  0.01823,   Adjusted R-squared:  0.008216 
F-statistic:  1.82 on 1 and 98 DF,  p-value: 0.1804</code></pre>
<pre class="r"><code>summary(lm(dat.sim$Z~dat.sim$x1))</code></pre>
<pre><code>
Call:
lm(formula = dat.sim$Z ~ dat.sim$x1)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.34617 -0.60943 -0.00449  0.54347  1.97565 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)  0.01487    0.08956   0.166   0.8684  
dat.sim$x1   0.18352    0.08010   2.291   0.0241 *
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.8936 on 98 degrees of freedom
Multiple R-squared:  0.05083,   Adjusted R-squared:  0.04115 
F-statistic: 5.249 on 1 and 98 DF,  p-value: 0.02411</code></pre>
</div>
<div id="simulation-with-two-latent-factors" class="section level3">
<h3>Simulation with two latent factors</h3>
<p>We added another factor that affects 5 traits (<span
class="math inline">\(Y_1,..., Y_5\)</span>). The parameters are set so
that <span class="math inline">\(PVE(Z\sim X)=2\%\)</span> and <span
class="math inline">\(PVE(Y\sim X) \approx 1.8\%\)</span></p>
<pre class="r"><code>sim_10trait2 &lt;- function(alpha1=0.1, alpha2=0.1 ,beta1=0.3, beta2=0.4, eps0=0.7, eps1=0.4, N=100){
  x = rnorm(N,sd=1)
  Z1 = alpha1*x + rnorm(N,sd=eps0)
  Z2 = alpha2*x + rnorm(N,sd=eps0)
  y = matrix(0,nrow = N,ncol=10)
  for(i in 1: ncol(y)){
    y[,i] &lt;- beta1*Z1 + rnorm(N,sd=eps1)
  }
  for(i in 1:3){ ### Add the effect of Z2
    y[,i] &lt;- y[,i] + beta2*Z2
  }
  list(x=x,Z=cbind(Z1,Z2),y=y)
}</code></pre>
<pre class="r"><code>library(pheatmap)
dat &lt;- sim_10trait2()
y1 &lt;- scale(dat$y)
pcs &lt;- prcomp(y1)
pheatmap(cor(pcs$x[,1:4], dat$Z)**2,cluster_cols = F,cluster_rows = F,
         display_numbers=round(cor(pcs$x[,1:4], dat$Z)**2,3),
         fontsize_number = 15, cellheight = 20, cellwidth=60)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/figure/factor_eqtl.Rmd/unnamed-chunk-12-1.png" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/ca49fde79ffca8148a0210108863787e554bb231/docs/figure/factor_eqtl.Rmd/unnamed-chunk-12-1.png" target="_blank">ca49fde</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/a9aaac7de843d66fdf7555bbf5eb74c2f0803465/docs/figure/factor_eqtl.Rmd/unnamed-chunk-12-1.png" target="_blank">a9aaac7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(pcs$sdev^2/sum(pcs$sdev^2),ylab=&quot;PVE&quot;)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/figure/factor_eqtl.Rmd/unnamed-chunk-13-1.png" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PCA results have captured one factor. My conjecture is that the two
factors are correlated by the genetic effect. So PCA tends to attribute
all genetic effects to the first PC.</p>
<pre class="r"><code>z_hat &lt;- pcs$x[,1]
plot(z_hat, dat$Z[,1], ylab=&quot;Oracle Z1&quot;, xlab=&quot; Estimated Z&quot;)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/figure/factor_eqtl.Rmd/unnamed-chunk-14-1.png" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/ca49fde79ffca8148a0210108863787e554bb231/docs/figure/factor_eqtl.Rmd/unnamed-chunk-14-1.png" target="_blank">ca49fde</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/a9aaac7de843d66fdf7555bbf5eb74c2f0803465/docs/figure/factor_eqtl.Rmd/unnamed-chunk-14-1.png" target="_blank">a9aaac7</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>z_hat &lt;- pcs$x[,2]
plot(z_hat, dat$Z[,2], ylab=&quot;Oracle Z2&quot;, xlab=&quot; Estimated Z&quot;)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/figure/factor_eqtl.Rmd/unnamed-chunk-15-1.png" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Nrep = 300

factor.res1 &lt;- matrix(nrow=Nrep,ncol=3)
oracle.z1 &lt;- matrix(nrow=Nrep,ncol=3)
factor.res2 &lt;- matrix(nrow=Nrep,ncol=3)
oracle.z2 &lt;- matrix(nrow=Nrep,ncol=3)
marg &lt;- list()
marg$beta &lt;- matrix(nrow=Nrep,ncol=10)
marg$tstat &lt;- matrix(nrow=Nrep,ncol=10)
marg$pval &lt;- matrix(nrow=Nrep,ncol=10)
  
for(i in 1:Nrep) {
  dat &lt;- sim_10trait2(N=100)
  pcs &lt;- prcomp(dat$y,center = T,scale. = T)
  z_hat &lt;- pcs$x[,1:2]
  factor.res1[i,] &lt;- summary(lm(z_hat[,1]~dat$x))$coefficients[2,c(1,3,4)]
  factor.res2[i,] &lt;- summary(lm(z_hat[,2]~dat$x))$coefficients[2,c(1,3,4)]
  oracle.z1[i,] &lt;- summary(lm(dat$Z[,1]~dat$x))$coefficients[2,c(1,3,4)]
  oracle.z2[i,] &lt;- summary(lm(dat$Z[,2]~dat$x))$coefficients[2,c(1,3,4)]
  #colnames(marg.res) &lt;- c(&quot;beta&quot;,&quot;tstat&quot;,&quot;p&quot;)
  #print(i)
  for(j in 1:ncol(dat$y)) {
    temp &lt;- summary(lm(dat$y[,j]~dat$x))$coefficients[2,]
    marg$beta[i,j] &lt;- temp[1]
    marg$tstat[i,j] &lt;- temp[3]
    marg$pval[i,j] &lt;- temp[4]
  }
}</code></pre>
</div>
<div id="frequency-of-rejecting-null-nominal-pv-0.05"
class="section level3">
<h3>Frequency of rejecting null (nominal PV &lt; 0.05)</h3>
<pre class="r"><code>dat2 &lt;- do.call(cbind, list(oracle_Z1=oracle.z1[,3],oracle_Z2=oracle.z2[,3],
                            Z1_hat=factor.res1[,3],Z2_hat=factor.res2[,3]))
dat2 &lt;- cbind(dat2, marg$pval)
colnames(dat2)[5:14] &lt;- paste0(&quot;trait &quot;,1:10)
barplot(colMeans(dat2&lt;0.05),las=2)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/f3f63288bc67e72bd8176f93871e24b790c7abb5/docs/figure/factor_eqtl.Rmd/unnamed-chunk-17-1.png" target="_blank">f3f6328</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-without-genotypes" class="section level2">
<h2>Simulation without genotypes</h2>
<p>In this case, the two factors are stastically independent from each
other.</p>
<pre class="r"><code>### simulation without X (genotypes)
sim_10trait3 &lt;- function(beta1=0.3, beta2=0.4, eps0=1.0, eps1=0.4, N=100){
  Z1 = rnorm(N,sd=eps0)
  Z2 = rnorm(N,sd=eps0)
  y = matrix(0,nrow = N,ncol=10)
  for(i in 1: ncol(y)){
    y[,i] &lt;- beta1*Z1 + rnorm(N,sd=eps1)
  }
  for(i in 1:3){ ### Add the effect of Z2
    y[,i] &lt;- y[,i] + beta2*Z2
  }
  list(Z=cbind(Z1,Z2),y=y)
}</code></pre>
<p>PCA results correlates with the two factors better.</p>
<pre class="r"><code>dat &lt;- sim_10trait3()
y1 &lt;- scale(dat$y)
pcs &lt;- prcomp(y1)
pheatmap(cor(pcs$x[,1:4], dat$Z)**2, cluster_cols = F, cluster_rows = F,
         display_numbers=round(cor(pcs$x[,1:4],dat$Z)**2,3),
         fontsize_number = 15, cellheight = 20, cellwidth=60)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>FastICA identified similar results but factors are usually more
distinguished.</p>
<pre class="r"><code>library(ica)

res &lt;- icafast(dat$y,2)
dat1 &lt;- cor(res$S,dat$Z)**2
pheatmap(dat1, cluster_cols = F, cluster_rows = F,
         display_numbers=round(dat1,3),
         fontsize_number = 15, cellheight = 20, cellwidth=60)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/8e34ef652943cc26d3fb8986a76a3329666674f4/docs/figure/factor_eqtl.Rmd/unnamed-chunk-20-1.png" target="_blank">8e34ef6</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="simulation-with-two-snps-affecting-two-factors-seperately"
class="section level2">
<h2>Simulation with two SNPs affecting two factors seperately</h2>
<pre class="r"><code>### simulation without 2 SNPs affecting two factors
sim_10trait4 &lt;- function(alpha1=0.1, alpha2=0.1, beta1=0.3, beta2=0.3, eps0=1.0, eps1=0.4, N=100){
  X1 = rnorm(N,sd=1)
  X2 = rnorm(N,sd=1)
  Z1 = alpha1*X1 + rnorm(N,sd=eps0)
  Z2 = alpha2*X2 + rnorm(N,sd=eps0)
  y = matrix(0,nrow = N,ncol=10)
  for(i in 1: ncol(y)){
    y[,i] &lt;- beta1*Z1 + rnorm(N,sd=eps1)
  }
  for(i in 1:3){ ### Add the effect of Z2
    y[,i] &lt;- y[,i] + beta2*Z2
  }
  list(x=cbind(X1,X2),Z=cbind(Z1,Z2),y=y)
}</code></pre>
<p>Heatmap below shows the <span class="math inline">\(R^2\)</span>
between PCs and oracle factor values. The first two PCs correlate well
with the oracle factor values.</p>
<pre class="r"><code>set.seed(20)
dat &lt;- sim_10trait4()
#y1 &lt;- scale(dat$y)
pcs &lt;- prcomp(dat$y,center = T,scale. = T)
pheatmap(cor(pcs$x[,1:4], dat$Z)**2, cluster_cols = F, cluster_rows = F,
         display_numbers=round(cor(pcs$x[,1:4],dat$Z)**2,3),
         fontsize_number = 15, cellheight = 20, cellwidth=60)</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/a6f5a6eaaa03120e60f3954a4026aaa8fca47a55/docs/figure/factor_eqtl.Rmd/unnamed-chunk-22-1.png" target="_blank">a6f5a6e</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/8e34ef652943cc26d3fb8986a76a3329666674f4/docs/figure/factor_eqtl.Rmd/unnamed-chunk-22-1.png" target="_blank">8e34ef6</a>
</td>
<td>
Lifan Liang
</td>
<td>
2025-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We repeated the simulation 300 times and check the statical
power.</p>
<pre class="r"><code>Nrep = 300

factor.res1 &lt;- matrix(nrow=Nrep,ncol=3)
oracle.z1 &lt;- matrix(nrow=Nrep,ncol=3)
factor.res2 &lt;- matrix(nrow=Nrep,ncol=3)
oracle.z2 &lt;- matrix(nrow=Nrep,ncol=3)
marg1 &lt;- list()
marg1$beta &lt;- matrix(nrow=Nrep,ncol=10)
marg1$tstat &lt;- matrix(nrow=Nrep,ncol=10)
marg1$pval &lt;- matrix(nrow=Nrep,ncol=10)
marg2 &lt;- marg1

for(i in 1:Nrep) {
  dat &lt;- sim_10trait4(N=100)
  pcs &lt;- prcomp(dat$y,center = T,scale. = T)
  z_hat &lt;- pcs$x[,1:2]
  factor.res1[i,] &lt;- summary(lm(z_hat[,1]~dat$x[,1]))$coefficients[2,c(1,3,4)]
  factor.res2[i,] &lt;- summary(lm(z_hat[,2]~dat$x[,2]))$coefficients[2,c(1,3,4)]
  oracle.z1[i,] &lt;- summary(lm(dat$Z[,1]~dat$x[,1]))$coefficients[2,c(1,3,4)]
  oracle.z2[i,] &lt;- summary(lm(dat$Z[,2]~dat$x[,2]))$coefficients[2,c(1,3,4)]
  #colnames(marg.res) &lt;- c(&quot;beta&quot;,&quot;tstat&quot;,&quot;p&quot;)
  #print(i)
  for(j in 1:ncol(dat$y)) {
    temp &lt;- summary(lm(dat$y[,j]~dat$x[,1]))$coefficients[2,]
    marg1$beta[i,j] &lt;- temp[1]
    marg1$tstat[i,j] &lt;- temp[3]
    marg1$pval[i,j] &lt;- temp[4]
    temp &lt;- summary(lm(dat$y[,j]~dat$x[,2]))$coefficients[2,]
    marg2$beta[i,j] &lt;- temp[1]
    marg2$tstat[i,j] &lt;- temp[3]
    marg2$pval[i,j] &lt;- temp[4]
  }
}</code></pre>
<pre class="r"><code>library(ggplot2)
dat2 &lt;- do.call(cbind, list(oracle_Z1=oracle.z1[,3],oracle_Z2=oracle.z2[,3],
                            Z1_hat=factor.res1[,3],Z2_hat=factor.res2[,3]))
dat2 &lt;- cbind(cbind(dat2, marg1$pval), marg2$pval)
colnames(dat2)[5:24] &lt;- paste0(&quot;trait &quot;,1:10)
dat3 &lt;- data.frame(frequency=colMeans(dat2&lt;0.05),
                   type=rep(c(&quot;Oracle&quot;,&quot;Estimated&quot;,&quot;Marginal&quot;),times=c(2,2,20)),
                   names=colnames(dat2))
dat3$names[c(1,3,5:14)] &lt;- paste(colnames(dat2)[c(1,3,5:14)],&quot;~&quot;,&quot;X1&quot;)
dat3$names[c(2,4,15:24)] &lt;- paste(colnames(dat2)[c(2,4,15:24)],&quot;~&quot;,&quot;X2&quot;)

ggplot(dat3,aes(x=names,y=frequency,fill=type)) + geom_bar(stat=&quot;identity&quot;) + 
  theme_bw() + theme(axis.text.x = element_text(angle = 45,hjust=1))</code></pre>
<p><img src="figure/factor_eqtl.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="eqtl-at-different-cellular-resolution" class="section level2">
<h2>eQTL at different cellular resolution</h2>
<p>I run eQTL mapping with three aggregation approach: (1) Take average
of expression across all cells per donor; (2) take average of expression
within the excitatory cell type (npglut and nmglut); (3) take average of
expression within inhibitory cell types (GABA). All eQTL mapping was run
with 5 genotype PCs, 15 expression PCs, and 95 donors. Total eQTL
identified 8072 eGenes; excitatory eQTL identified 6926 eGenes;
inhibitory eQTL identified 5155 eGenes.</p>
<p><img src="assets/psbulk_eqtl_overlap.png" /></p>
<div id="comparing-with-eqtl-within-single-context"
class="section level3">
<h3>Comparing with eQTL within single context</h3>
<p><img src="assets/gaba1_inh_total.png" /></p>
<p><img src="assets/npglut1_exc_total.png" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ica_1.0-2       pheatmap_1.0.12 flashier_1.0.7  magrittr_2.0.2 
[5] ebnm_1.1-2      ggplot2_3.3.5   workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.11        horseshoe_0.2.0    invgamma_1.1       lattice_0.20-45   
 [5] tidyr_1.2.0        getPass_0.2-2      ps_1.6.0           assertthat_0.2.1  
 [9] rprojroot_2.0.2    digest_0.6.29      utf8_1.2.2         truncnorm_1.0-9   
[13] R6_2.5.1           evaluate_0.14      httr_1.4.2         highr_0.9         
[17] pillar_1.7.0       rlang_1.1.6        rstudioapi_0.13    irlba_2.3.5       
[21] whisker_0.4        callr_3.7.0        jquerylib_0.1.4    Matrix_1.3-4      
[25] rmarkdown_2.11     labeling_0.4.2     splines_4.1.2      stringr_1.4.0     
[29] munsell_0.5.0      mixsqp_0.3-43      compiler_4.1.2     httpuv_1.6.5      
[33] xfun_0.29          pkgconfig_2.0.3    SQUAREM_2021.1     htmltools_0.5.2   
[37] tidyselect_1.1.1   tibble_3.1.6       fansi_1.0.2        crayon_1.4.2      
[41] dplyr_1.0.7        withr_2.4.3        later_1.3.0        grid_4.1.2        
[45] jsonlite_1.7.3     gtable_0.3.0       lifecycle_1.0.4    DBI_1.1.2         
[49] git2r_0.29.0       scales_1.1.1       cli_3.6.5          stringi_1.7.6     
[53] farver_2.1.0       fs_1.5.2           promises_1.2.0.1   bslib_0.3.1       
[57] ellipsis_0.3.2     generics_0.1.2     vctrs_0.6.5        trust_0.1-8       
[61] RColorBrewer_1.1-2 tools_4.1.2        softImpute_1.4-1   glue_1.6.1        
[65] purrr_0.3.4        processx_3.5.2     parallel_4.1.2     fastmap_1.1.0     
[69] yaml_2.2.2         colorspace_2.0-2   ashr_2.2-54        deconvolveR_1.2-1 
[73] knitr_1.37         sass_0.4.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
