<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lifan Liang" />

<meta name="date" content="2026-01-20" />

<title>mashR_extension</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">eQTL_methods</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">mashR_extension</h1>
<h4 class="author">Lifan Liang</h4>
<h4 class="date">2026-01-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>eQTL_methods/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250106code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250106)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250106code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250106)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstree0c4495218ed96cea4b629bd493cf6d446d456eabtargetblank0c44952a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LifanLiang/eQTL_methods/tree/0c4495218ed96cea4b629bd493cf6d446d456eab" target="_blank">0c44952</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLifanLiangeQTLmethodstree0c4495218ed96cea4b629bd493cf6d446d456eabtargetblank0c44952a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LifanLiang/eQTL_methods/tree/0c4495218ed96cea4b629bd493cf6d446d456eab" target="_blank">0c44952</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory

Untracked files:
    Untracked:  data/neuron_stim_genotype_chr9_2e6-3e6.rds
    Untracked:  data/susie_code.R
    Untracked:  data/susie_simulation.zip
    Untracked:  test_matrixeqtl.R

Unstaged changes:
    Modified:   data/mixedmodel_res_phi0.02.csv
    Modified:   data/mixedmodel_res_phi0.03.csv
    Modified:   data/mixedmodel_res_phi0.04.csv
    Modified:   data/mixedmodel_res_phi0.05.csv
    Modified:   data/scalability_snp.csv
    Modified:   data/twostep_res_phi0.02.csv
    Modified:   data/twostep_res_phi0.03.csv
    Modified:   data/twostep_res_phi0.04.csv
    Modified:   data/twostep_res_phi0.05.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mashR_extension.Rmd</code>) and
HTML (<code>docs/mashR_extension.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/0c4495218ed96cea4b629bd493cf6d446d456eab/analysis/mashR_extension.Rmd" target="_blank">0c44952</a>
</td>
<td>
Lifan Liang
</td>
<td>
2026-01-26
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LifanLiang/eQTL_methods/b42d3741d22acf8eb259d1707be2ff119598bf3d/docs/mashR_extension.html" target="_blank">b42d374</a>
</td>
<td>
Lifan Liang
</td>
<td>
2026-01-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/03517e0ba31a215595f83999605ab7f0408df75c/analysis/mashR_extension.Rmd" target="_blank">03517e0</a>
</td>
<td>
Lifan Liang
</td>
<td>
2026-01-21
</td>
<td>
wflow_publish("analysis")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
.scroll-300 {
  max-height: 300px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div id="simulation-experiment" class="section level2">
<h2>Simulation experiment</h2>
<div id="generative-process-for-eqtl-in-two-contexts"
class="section level3">
<h3>Generative process for eQTL in two contexts</h3>
<ul>
<li><p><span class="math inline">\(\pi_0\)</span>: the probability that
genetic effects sharing is zero (<span
class="math inline">\(\rho_G=0\)</span>)</p></li>
<li><p>Z: the intermediate binary variable indicating whether the
genetic effects are independent or not.</p></li>
<li><p><span class="math inline">\(G\)</span>: a one-dimensional vector
of genotypes sampled from standard Normal. The two contexts shared the
same set of donors, therefore the same set of genotype vector.</p></li>
<li><p><span class="math inline">\(\beta\)</span>: a vector for genetic
effects sampled from a multivariate Normal (MVN). <span
class="math inline">\(\beta_1\)</span> is the genetic effect in the
first context, and <span class="math inline">\(\beta_2\)</span> is the
genetic for the second context.</p></li>
<li><p><span class="math inline">\(\sigma_G\)</span>: the standard
deviation of genetic effects. Can be called the effect size. The two
contexts have the same magnitude of effect size.</p></li>
<li><p><span class="math inline">\(\rho_G\)</span>: the correlation of
genetic effects between the two contexts.</p></li>
<li><p><span class="math inline">\(\sigma_e\)</span>: the standard
deviation of environmental effects (include measure noise/error?). The
two contexts have the same magnitude of environmental effect
size.</p></li>
<li><p><span class="math inline">\(\rho_e\)</span>: the correlation of
environmenta effects between the two contexts.</p></li>
<li><p><span class="math inline">\(Y\)</span>: a vector of gene
expression</p></li>
</ul>
<p><span class="math display">\[
Z \sim Bernoulli(1-\pi_0)
\]</span></p>
<p><span class="math display">\[
Z = \begin{cases}
1 &amp; \rho_G&gt;0 \\
0 &amp; \rho_G=0
\end{cases}
\]</span></p>
<p><span class="math display">\[
G \sim Normal(0,1)
\]</span></p>
<p><span class="math display">\[
\beta \sim MVN(0,\begin{bmatrix}
1 &amp; \rho_G  \\
\rho_G &amp; 1
\end{bmatrix} \cdot \sigma_G^2)
\]</span></p>
<p><span class="math display">\[
\epsilon \sim MVN(0, \begin{bmatrix}
1 &amp; \rho_e  \\
\rho_e &amp; 1
\end{bmatrix} \sigma_e^2)
\]</span></p>
<p><span class="math display">\[
Y_1 = G \beta_1 + \epsilon_1
\]</span></p>
<p><span class="math display">\[
Y_2 = G \beta_2 + \epsilon_2
\]</span></p>
<p>The estimate of genetic effects are generated from ordinary least
square (OLS):</p>
<p><span class="math display">\[
\hat\beta_1 = (G^T G)^{-1}G^T Y_1
\]</span></p>
<p><span class="math display">\[
\hat\beta_2 = (G^T G)^{-1}G^T Y_2
\]</span></p>
<p>We repeat the procedure above for 100 times, each time with <span
class="math inline">\(n\)</span> donors (by default <span
class="math inline">\(n=100\)</span>). And we obtained 100 estimated
genetic effects from the two contexts.</p>
<pre class="r scroll-300"><code>library(MASS)

n &lt;- 50
m &lt;- 10000
sigma_G2 &lt;- 0.1
sigma_e2 &lt;- 1 - sigma_G2
rho_G &lt;- 0.2

G &lt;- matrix(rep(rnorm(n),m), nrow = n, ncol = m)


sim_marker_regression &lt;- function(n = 100, m = 10, sigma_G2 = 0.1, sigma_e2 = 1.0, rho_G = 0.3, rho_e=0.9, G=G) {
  
  # 1. Generate True Betas (Genetic Effects)
  # Correlation between trait 1 and trait 2 is 0.5 as per your prompt
  U &lt;- matrix(c(1, rho_G, rho_G, 1), nrow = 2)
  cov_beta &lt;- U * sigma_G2
  betas_true &lt;- mvrnorm(n = m, mu = c(0, 0), Sigma = cov_beta)
  colnames(betas_true) &lt;- c(&quot;Beta1_True&quot;, &quot;Beta2_True&quot;)
  
  # 2. Generate Residuals (Environmental Effects)
  cov_epsilon &lt;- matrix(c(1, rho_e, rho_e, 1), nrow = 2) * sigma_e2
  eps_all &lt;- mvrnorm(n = n * m, mu = c(0, 0), Sigma = cov_epsilon)
  
  # Reshape residuals into n x m matrices
  eps1_mat &lt;- matrix(eps_all[, 1], nrow = n, ncol = m)
  eps2_mat &lt;- matrix(eps_all[, 2], nrow = n, ncol = m)
  
  # 3. Generate Phenotype Matrices Y1 and Y2 (n x m)
  Y1 &lt;- sweep(G, 2, betas_true[, 1], `*`) + eps1_mat
  Y2 &lt;- sweep(G, 2, betas_true[, 2], `*`) + eps2_mat
  
  # 4. Regression (OLS estimation)
  # Using the fast matrix method: beta_hat = (G&#39;G)^-1 G&#39;Y
  beta1_hat &lt;- colSums(G * Y1) / colSums(G^2)
  beta2_hat &lt;- colSums(G * Y2) / colSums(G^2)
  
  # 5. Consolidate Results
  results &lt;- data.frame(
    marker_id = 1:m,
    beta1_true = betas_true[, 1],
    beta1_est  = beta1_hat,
    beta2_true = betas_true[, 2],
    beta2_est  = beta2_hat
  )
  
  return(results)
}</code></pre>
</div>
<div
id="how-environmental-effect-sharing-leaks-into-genetic-effect-sharing."
class="section level3">
<h3>How environmental effect sharing leaks into genetic effect
sharing.</h3>
<p>When samples are completely overlapped, the sharing of environmental
effects (residual correlation) will leak into the estimate of sharing of
genetic effects.</p>
<p><span class="math display">\[
cov(\hat\beta_1, \hat\beta_2) = \rho_G \sigma_G^2 + \dfrac{\rho_e
\sigma_e^2}{G^T G}
\]</span></p>
<p>As derived above, <span class="math inline">\(\rho_G
\sigma_G^2\)</span> represents the true genetic effect sharing. <span
class="math inline">\(\dfrac{\rho_e \sigma_e^2}{G^T G}\)</span> is the
proportion of residual correlation added into the estimated sharing of
genetic effects. Since is <span class="math inline">\(G\)</span> is from
standard Normal in our case and usually normalized in practice, we can
simply approximate the environmental proportion as <span
class="math inline">\(\dfrac{\rho_e \sigma_e^2}{n}\)</span>.</p>
<p>Please note that the derivation assumes <span
class="math inline">\(G\)</span> is not a random variable. Hence, all
the repeated simulations uses the same <span
class="math inline">\(G\)</span>.</p>
<p>As shown in the figure below, blue dots is observed covariance of
genetic effect along with different residual correlation. The green line
indicates the theoretical trend (similar to linear regression: <span
class="math inline">\(\hat\rho_G=\rho_G \sigma_G^2 +
\dfrac{\sigma_e^2}{\sum G_i^2}\cdot \rho_e\)</span>. And the red line is
the true genetic covariance along)</p>
<pre class="r scroll-300"><code>set.seed(1234)

library(MASS)
library(ggplot2)

n &lt;- 50
m &lt;- 10000
sigma_G2 &lt;- 0.1
sigma_e2 &lt;- 1 - sigma_G2
rho_G &lt;- 0.2

rho_e_values &lt;- seq(0, 0.95, by = 0.05)
estimated_corrs &lt;- numeric(length(rho_e_values))

G &lt;- matrix(rep(rnorm(n),m), nrow = n, ncol = m) # Replicate G for M times for vectorized computation
#G &lt;- matrix(rnorm(n*m), nrow = n, ncol = m)

for (i in seq_along(rho_e_values)) {
  sim_res &lt;- sim_marker_regression(
    n = n, 
    m = m, 
    sigma_G2 = sigma_G2, 
    sigma_e2 = sigma_e2, 
    rho_G = rho_G, 
    rho_e = rho_e_values[i],
    G = G
  )
  estimated_corrs[i] &lt;- cov(sim_res$beta1_est, sim_res$beta2_est)
}

plot_data &lt;- data.frame(rho_e = rho_e_values, est_corr = estimated_corrs)

g_sum_sq &lt;- sum(G[,1]^2) 

theoretical_intercept &lt;- rho_G * sigma_G2
theoretical_slope     &lt;- sigma_e2 / g_sum_sq

ggplot(plot_data, aes(x = rho_e, y = est_corr)) +
  ylim(c(0,0.042)) + 
  geom_line(color = &quot;steelblue&quot;, size = 1) +
  geom_point(color = &quot;darkblue&quot;) +
  geom_hline(yintercept = rho_G*sigma_G2, linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
  geom_abline(
    intercept = theoretical_intercept, 
    slope = theoretical_slope, 
    color = &quot;green&quot;, 
    linetype = &quot;dashed&quot;, 
    size = 1
  ) +
  annotate(&quot;text&quot;, x = 0.25, y = rho_G*sigma_G2-2e-3, label = paste0(&quot;True Genetic covariance&quot;,rho_G*sigma_G2), color = &quot;red&quot;) +
  labs(
    subtitle = paste0(n, &quot; samples, &quot;,m, &quot; repeats, sigma_G2=&quot;, sigma_G2, &quot;, rho_G=&quot;, rho_G),
    x = &quot;Residual Correlation&quot;,
    y = &quot;Covariance of estimated genetic effects between two contexts&quot;
  ) +
  theme_minimal()</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LifanLiang/eQTL_methods/blob/b42d3741d22acf8eb259d1707be2ff119598bf3d/docs/figure/mashR_extension.Rmd/unnamed-chunk-2-1.png" target="_blank">b42d374</a>
</td>
<td>
Lifan Liang
</td>
<td>
2026-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="posterior-of-genetic-effects-overly-corrects-the-confounding-of-residual-correlation"
class="section level3">
<h3>Posterior of genetic effects overly corrects the confounding of
residual correlation</h3>
<p>Given the prior distribution of true genetic effects and the
estimated genetic effects, we recovered the posterior of genetic effects
between two contexts and compared how LFSR changes according to the
specified residual correlation when the true residual correlation is
high (0.9).</p>
<p>Following the MASH paper, the posterior of true genetic effect (<span
class="math inline">\(\tilde\beta\)</span>) can be derived from the
estimated genetic effect (<span
class="math inline">\(\hat\beta\)</span>), prior covariance matrix
(<span class="math inline">\(U=\begin{bmatrix} 1 &amp; \rho_G \\ \rho_G
&amp; 1 \end{bmatrix} \cdot \sigma_G^2\)</span>), and the residual
correlation (<span class="math inline">\(V=\begin{bmatrix} 1 &amp;
\rho_e \\ \rho_e &amp; 1 \end{bmatrix} \cdot
\dfrac{\sigma_e^2}{n}\)</span>).</p>
<p><span class="math display">\[
\Sigma_{post} = (U^{-1} + V^{-1})^{-1}
\]</span></p>
<p><span class="math display">\[
E(\tilde\beta|\hat\beta) = \Sigma_{post} V^{-1} \hat{\beta}
\]</span></p>
<p>As a sanity check, we compared the accuracy (correlation with ground
truth) of OLS estimator and the posterior mean. As <span
class="math inline">\(\rho_e\)</span> increased, posterior mean have
large accuracy gain.</p>
<pre class="r"><code>library(MASS)

compute_posterior &lt;- function(U, V, beta_hat) {
  Total_Variance &lt;- V + U
  W &lt;- U %*% solve(Total_Variance)

  mu_post &lt;- W %*% beta_hat
  Sigma_post &lt;- U - (W %*% U)
  
  return(list(mu_post = mu_post, Sigma_post = Sigma_post))
}</code></pre>
<pre class="r scroll-300"><code>set.seed(1234)
n &lt;- 50
m &lt;- 10000
sigma_G2 &lt;- 0.1
sigma_e2 &lt;- 1 - sigma_G2  # Assuming total variance is 1
rho_G &lt;- 0.2

rho_e_values &lt;- seq(0, 0.95, by = 0.05)

ols_corrs &lt;- numeric(length(rho_e_values))
post_corrs &lt;- numeric(length(rho_e_values))

G &lt;- matrix(rep(rnorm(n), m), nrow = n, ncol = m)
g_sum_sq &lt;- sum(G[,1]^2) # The scaling factor for error variance

# Vectors to store accuracy correlations
acc_ols &lt;- numeric(length(rho_e_values))
acc_post &lt;- numeric(length(rho_e_values))

for (i in seq_along(rho_e_values)) {
  rho_e_current &lt;- rho_e_values[i]

  sim_res &lt;- sim_marker_regression(
    n = n, m = m, sigma_G2 = sigma_G2, sigma_e2 = sigma_e2, 
    rho_G = rho_G, rho_e = rho_e_current, G = G
  )
  ols_corrs[i] &lt;- cov(sim_res$beta1_est, sim_res$beta2_est)
  
  U &lt;- matrix(c(sigma_G2, rho_G*sigma_G2, rho_G*sigma_G2, sigma_G2), 2, 2)
  
  scaling &lt;- 1 / g_sum_sq
  V &lt;- matrix(c(sigma_e2, rho_e_current*sigma_e2, 
                rho_e_current*sigma_e2, sigma_e2), 2, 2) * scaling

  beta_hat_matrix &lt;- rbind(sim_res$beta1_est, sim_res$beta2_est)

  post_res &lt;- compute_posterior(U, V, beta_hat_matrix)
  
  post_corrs[i] &lt;- cov(post_res$mu_post[1, ], post_res$mu_post[2, ])

  acc_ols[i] &lt;- cor(sim_res$beta1_est, sim_res$beta1_true)

  acc_post[i] &lt;- cor(post_res$mu_post[1, ], sim_res$beta1_true)
}

plot_data &lt;- data.frame(
  rho_e = rho_e_values,
  OLS_Accuracy = acc_ols,
  Posterior_Accuracy = acc_post
)

ggplot(plot_data, aes(x = rho_e)) +
  geom_line(aes(y = OLS_Accuracy, color = &quot;OLS Estimate (beta_hat)&quot;), size = 1) +
  geom_point(aes(y = OLS_Accuracy, color = &quot;OLS Estimate (beta_hat)&quot;)) +
  geom_line(aes(y = Posterior_Accuracy, color = &quot;Posterior Mean&quot;), size = 1) +
  geom_point(aes(y = Posterior_Accuracy, color = &quot;Posterior Mean&quot;)) +
  scale_color_manual(values = c(&quot;steelblue&quot;, &quot;orange&quot;)) +
  labs(
    title = &quot;Estimation Accuracy: OLS vs Posterior&quot;,
    y = &quot;Correlation with Ground Truth (True Beta)&quot;,
    x = &quot;Environmental Correlation (rho_e)&quot;
  ) +
  theme_minimal()</code></pre>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>However, the covariance of posterior mean will overcorrect the
residual correlation as <span class="math inline">\(\rho_e\)</span>
increased. This is due to the law of total variance:</p>
<p><span class="math display">\[
cov(\beta) = E[Var(\beta|\hat\beta)] + Var[E(\beta|\hat\beta)]
\]</span></p>
<p>which can be rewritten as:</p>
<p><span class="math display">\[
\rho_G \sigma_G^2= \Sigma_{post} + cov(\tilde\beta)
\]</span></p>
<p><span class="math inline">\(\Sigma_{post}\)</span> will get larger as
<span class="math inline">\(\rho_e\)</span> increased. Since the prior
covariance of genetic effects is constant, the increase of <span
class="math inline">\(\Sigma_post\)</span> leads to the decrease of
<span class="math inline">\(cov(\tilde\beta)\)</span>.</p>
<pre class="r scroll-300"><code>plot_data &lt;- data.frame(
  rho_e = rho_e_values, 
  OLS_Cov = ols_corrs,
  Post_Cov = post_corrs
)

theoretical_intercept &lt;- rho_G * sigma_G2
theoretical_slope     &lt;- sigma_e2 / g_sum_sq

ggplot(plot_data, aes(x = rho_e)) +
  # OLS Lines
  geom_line(aes(y = OLS_Cov, color = &quot;OLS Estimate&quot;), size = 1) +
  geom_point(aes(y = OLS_Cov, color = &quot;OLS Estimate&quot;)) +
  
  # Posterior Lines
  geom_line(aes(y = Post_Cov, color = &quot;Posterior Mean&quot;), size = 1, linetype=&quot;solid&quot;) +
  geom_point(aes(y = Post_Cov, color = &quot;Posterior Mean&quot;)) +
  
  # Theoretical OLS Slope (Green Dashed)
  geom_abline(intercept = theoretical_intercept, slope = theoretical_slope, 
              color = &quot;green&quot;, linetype = &quot;dashed&quot;, size = 0.8) +
  
  # True Genetic Covariance (Red Dashed)
  geom_hline(yintercept = rho_G*sigma_G2, linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
  
  annotate(&quot;text&quot;, x = 0.25, y = rho_G*sigma_G2 - 0.002, 
           label = paste0(&quot;True Genetic Cov (&quot;, rho_G*sigma_G2, &quot;)&quot;), color = &quot;red&quot;) +
  
  scale_color_manual(name = &quot;Method&quot;, 
                     values = c(&quot;OLS Estimate&quot; = &quot;steelblue&quot;, &quot;Posterior Mean&quot; = &quot;orange&quot;)) +
  
  labs(
    subtitle = paste0(n, &quot; samples, &quot;, m, &quot; repeats.&quot;),
    x = &quot;Residual Correlation (rho_e)&quot;,
    y = &quot;Covariance of Effects Between Two Contexts&quot;
  ) +
  theme_minimal()</code></pre>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="lfsr-inference-overestimates-the-genetic-effect-sharing"
class="section level3">
<h3>LFSR inference overestimates the genetic effect sharing</h3>
<p>To determine how LFSR was inflated, the parameters between data
generation and posterior inference are very different.</p>
<p>During the data generation, we set <span
class="math inline">\(\rho_G=0\)</span> and <span
class="math inline">\(sigma_G^2=0\)</span>. The true <span
class="math inline">\(\rho_e\)</span> go through a range of values from
0 to 0.95 to generate 20 different datasets.</p>
<p>During posterior inference, <span
class="math inline">\(\rho_e\)</span> was always set. <span
class="math inline">\(\sigma_G^2=1\)</span> will cause inflation to an
extent. But it also reduce the posterior weights on the prior. Hence
<span class="math inline">\(\sigma_G^2\)</span> does not affect the
results too much. The key setting is the prior covariance of genetic
effect (<span class="math inline">\(\rho_G\)</span>). Before introducing
a mixture of two settings of <span
class="math inline">\(\rho_G\)</span>, we need to validate the impact of
<span class="math inline">\(\rho_e\)</span> in the scenario that <span
class="math inline">\(\rho_G=0\)</span> and <span
class="math inline">\(\rho_G&gt;0\)</span> seperately.</p>
<p>We use LFSR&lt;5% as significant and computed the empirical False
discovery rates for each value of true <span
class="math inline">\(\rho_e\)</span>.</p>
<div id="rho_g0" class="section level4">
<h4><span class="math inline">\(\rho_G=0\)</span></h4>
<pre class="r"><code># Simply marginal posterior to compute LFSR
get_lfsr &lt;- function(post_res) {
  prob_less_0_1 &lt;- pnorm(0, mean = post_res$mu_post[1,], sd = sqrt(post_res$Sigma_post[1,1]))
  lfsr1 &lt;- pmin(prob_less_0_1, 1 - prob_less_0_1)
  
  prob_less_0_2 &lt;- pnorm(0, mean = post_res$mu_post[2,], sd = sqrt(post_res$Sigma_post[2,2]))
  lfsr2 &lt;- pmin(prob_less_0_2, 1 - prob_less_0_2)
  
  cbind(lfsr1, lfsr2)
}</code></pre>
<pre class="r scroll-300"><code>rho_e_hat_fixed &lt;- 0 # The model will ALWAYS assume independence (Naive)
rho_e_true_values &lt;- seq(0, 0.95, by = 0.05) # The data gets progressively more correlated

marginal_fpr &lt;- numeric(length(rho_e_true_values))
joint_fpr    &lt;- numeric(length(rho_e_true_values))

for (i in seq_along(rho_e_true_values)) {
  rho_current &lt;- rho_e_true_values[i]

  sim_res &lt;- sim_marker_regression(
    n = n, m = m, sigma_G2 = 0, sigma_e2 = 1, 
    rho_G = 0, rho_e = rho_current, G = G
  )
  
  U_naive &lt;- diag(1, 2)  # Causes inflation but also reduce the weight of the prior
  
  scaling &lt;- 1 / g_sum_sq
  V_naive &lt;- diag(sigma_e2, 2) * scaling
  
  beta_hat_matrix &lt;- rbind(sim_res$beta1_est, sim_res$beta2_est)

  post_res &lt;- compute_posterior(U_naive, V_naive, beta_hat_matrix)
  lfsr &lt;- get_lfsr(post_res)

  marginal_fpr[i] &lt;- mean(lfsr[,1] &lt; 0.05)

  joint_fpr[i] &lt;- mean(lfsr[,1] &lt; 0.05 &amp; lfsr[,2] &lt; 0.05)
}</code></pre>
<p>Given that <span class="math inline">\(\rho_G=0\)</span>, the
marginal FPR was inflated because of positive covariance in the prior.
The marignal LFSR was not affected by <span
class="math inline">\(\rho_e\)</span>. However, the proportion of LFSR
being significant in the two contexts together will grow as <span
class="math inline">\(rho_e\)</span> increased.</p>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="rho_g0.9" class="section level4">
<h4><span class="math inline">\(\rho_G=0.9\)</span></h4>
<pre class="r scroll-300"><code>rho_e_hat_fixed &lt;- 0 # The model will ALWAYS assume independence (Naive)
rho_e_true_values &lt;- seq(0, 0.95, by = 0.05) # The data gets progressively more correlated

marginal_fpr &lt;- numeric(length(rho_e_true_values))
joint_fpr    &lt;- numeric(length(rho_e_true_values))

for (i in seq_along(rho_e_true_values)) {
  rho_current &lt;- rho_e_true_values[i]

  sim_res &lt;- sim_marker_regression(
    n = n, m = m, sigma_G2 = 0, sigma_e2 = 1, 
    rho_G = 0, rho_e = rho_current, G = G
  )
  
  U_naive &lt;- matrix(c(1,0.9,0.9,1),2,2)  # Causes inflation but also reduce the weight of the prior
  
  scaling &lt;- 1 / g_sum_sq
  V_naive &lt;- diag(sigma_e2, 2) * scaling
  
  beta_hat_matrix &lt;- rbind(sim_res$beta1_est, sim_res$beta2_est)

  post_res &lt;- compute_posterior(U_naive, V_naive, beta_hat_matrix)
  lfsr &lt;- get_lfsr(post_res)

  marginal_fpr[i] &lt;- mean(lfsr[,1] &lt; 0.05)

  joint_fpr[i] &lt;- mean(lfsr[,1] &lt; 0.05 &amp; lfsr[,2] &lt; 0.05)
}</code></pre>
<p>Even when prior correlation was set to 0.9. Marginal LFSR only
slightly increased as <span class="math inline">\(\rho_e\)</span>
increased. But the phenomenon of joint false positive becomes more
obvious in this scenario.</p>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="a-mixture-of-rho_g0-and-rho_g0.9" class="section level4">
<h4>A mixture of <span class="math inline">\(\rho_G=0\)</span> and <span
class="math inline">\(\rho_G=0.9\)</span></h4>
<p>Combining the two scenarios above, we estimate the posterior with a
mixture of two scenarios: (1) genetic effects are independent between
contexts (<span class="math inline">\(\pi_0=0.1, \rho_G=0\)</span>); (2)
genetic effects are highly shared between the two contexts (<span
class="math inline">\(\pi_0=0.9, \rho_G=0.9\)</span>)</p>
<p>With the increase of <span class="math inline">\(\rho_e\)</span>, the
marginal FPR slightly increased, and the proportion of joint FPR
(simultaneous false postives in the two contexts) increased
substantially.</p>
<pre class="r scroll-300"><code>library(MASS)
library(ggplot2)
library(mvtnorm) 

compute_mixture_lfsr &lt;- function(beta_hat_matrix, V_naive, pi_weights, U_list) {
  # beta_hat_matrix: 2 x m
  # V_naive: 2 x 2 (Likelihood Error)
  # pi_weights: Vector of prior weights (e.g., c(0.1, 0.9))
  # U_list: List of Prior Covariance Matrices
  
  m &lt;- ncol(beta_hat_matrix)
  K &lt;- length(U_list)
  
  # Storage for component-wise stats
  log_lik_matrix &lt;- matrix(0, nrow = m, ncol = K)
  comp_prob_less_0_trait1 &lt;- matrix(0, nrow = m, ncol = K)
  comp_prob_less_0_trait2 &lt;- matrix(0, nrow = m, ncol = K)
  
  # Loop through Mixture Components
  for (k in 1:K) {
    U_k &lt;- U_list[[k]]

    Sigma_D &lt;- U_k + V_naive
    
    log_lik_matrix[, k] &lt;- dmvnorm(t(beta_hat_matrix), mean = c(0,0), sigma = Sigma_D, log = TRUE)
    
    W_k &lt;- U_k %*% solve(Sigma_D)
    mu_post_k &lt;- W_k %*% beta_hat_matrix # 2 x m
    Sigma_post_k &lt;- U_k - (W_k %*% U_k)

    sd_post_k &lt;- sqrt(diag(Sigma_post_k))

    comp_prob_less_0_trait1[, k] &lt;- pnorm(0, mean = mu_post_k[1,], sd = sd_post_k[1])
    comp_prob_less_0_trait2[, k] &lt;- pnorm(0, mean = mu_post_k[2,], sd = sd_post_k[2])
  }
  
  # exp(log_lik + log_prior - max_log_lik) trick for numerical stability
  log_weighted_lik &lt;- log_lik_matrix + matrix(log(pi_weights), nrow=m, ncol=K, byrow=TRUE)
  row_max &lt;- apply(log_weighted_lik, 1, max)
  exp_term &lt;- exp(log_weighted_lik - row_max)
  gammas &lt;- exp_term / rowSums(exp_term) # m x K matrix of posterior weights

  mix_prob_less_0_t1 &lt;- rowSums(gammas * comp_prob_less_0_trait1)
  mix_prob_less_0_t2 &lt;- rowSums(gammas * comp_prob_less_0_trait2)
  
  lfsr1 &lt;- pmin(mix_prob_less_0_t1, 1 - mix_prob_less_0_t1)
  lfsr2 &lt;- pmin(mix_prob_less_0_t2, 1 - mix_prob_less_0_t2)
  
  return(cbind(lfsr1, lfsr2))
}

set.seed(42)
n &lt;- 50
m &lt;- 2000 # Number of markers
sigma_e2 &lt;- 1
rho_e_true_values &lt;- seq(0, 0.95, by = 0.05)

G &lt;- matrix(rep(rnorm(n), m), nrow = n, ncol = m)
g_sum_sq &lt;- sum(G[,1]^2)
scaling &lt;- 1 / g_sum_sq

U_ind &lt;- diag(1, 2)
U_shared &lt;- matrix(c(1, 0.9, 0.9, 1), 2, 2)

U_list &lt;- list(U_ind, U_shared)
pi_weights &lt;- c(0.1, 0.9) # 90% belief in shared effects

# Storage
res_marginal &lt;- numeric(length(rho_e_true_values))
res_joint &lt;- numeric(length(rho_e_true_values))

for (i in seq_along(rho_e_true_values)) {
  rho_current &lt;- rho_e_true_values[i]
  
  # A. Simulate Null Data with True Correlation
  cov_e &lt;- matrix(c(1, rho_current, rho_current, 1), 2, 2)
  eps &lt;- mvrnorm(n*m, c(0,0), cov_e)
  Y1 &lt;- matrix(eps[,1], n, m); Y2 &lt;- matrix(eps[,2], n, m)
  denom &lt;- colSums(G^2)
  beta_hat &lt;- rbind(colSums(G*Y1)/denom, colSums(G*Y2)/denom)
  
  # B. Naive Likelihood (ignorant of residual correlation)
  V_naive &lt;- diag(1, 2) * scaling
  
  # C. Run Mixture Inference
  lfsr_mix &lt;- compute_mixture_lfsr(beta_hat, V_naive, pi_weights, U_list)
  
  # D. Calculate FPRs (Threshold 0.05)
  res_marginal[i] &lt;- mean(lfsr_mix[,1] &lt; 0.05)
  res_joint[i]    &lt;- mean(lfsr_mix[,1] &lt; 0.05 &amp; lfsr_mix[,2] &lt; 0.05)
}

# --- 4. Plotting ---
plot_data &lt;- data.frame(
  rho_e = rho_e_true_values,
  Marginal = res_marginal,
  Joint = res_joint
)

ggplot(plot_data, aes(x = rho_e)) +
  geom_line(aes(y = Marginal, color = &quot;Marginal FPR (Trait 1)&quot;), size = 1) +
  geom_point(aes(y = Marginal, color = &quot;Marginal FPR (Trait 1)&quot;)) +
  
  geom_line(aes(y = Joint, color = &quot;Joint FPR (Both)&quot;), size = 1) +
  geom_point(aes(y = Joint, color = &quot;Joint FPR (Both)&quot;)) +
  
  scale_color_manual(values = c(&quot;red&quot;, &quot;steelblue&quot;)) +
  labs(
    x = &quot;True Environmental Correlation (rho_e)&quot;,
    y = &quot;False Positive Rate (FPR)&quot;
  ) +
  theme_minimal()+
  theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figure/mashR_extension.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mvtnorm_1.1-3   ggplot2_4.0.1   MASS_7.3-54     workflowr_1.7.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10        RColorBrewer_1.1-3 compiler_4.1.2     pillar_1.11.1     
 [5] bslib_0.3.1        later_1.3.1        git2r_0.32.0       jquerylib_0.1.4   
 [9] tools_4.1.2        getPass_0.2-4      digest_0.6.31      gtable_0.3.6      
[13] jsonlite_2.0.0     evaluate_1.0.5     lifecycle_1.0.4    tibble_3.3.0      
[17] pkgconfig_2.0.3    rlang_1.1.6        cli_3.6.5          rstudioapi_0.17.1 
[21] yaml_2.3.11        xfun_0.54          fastmap_1.1.1      withr_3.0.2       
[25] dplyr_1.1.2        httr_1.4.7         stringr_1.6.0      knitr_1.50        
[29] generics_0.1.4     fs_1.6.2           vctrs_0.6.5        sass_0.4.6        
[33] tidyselect_1.2.1   grid_4.1.2         rprojroot_2.1.1    glue_1.8.0        
[37] R6_2.6.1           processx_3.8.6     otel_0.2.0         rmarkdown_2.30    
[41] farver_2.1.1       callr_3.7.6        magrittr_2.0.4     whisker_0.4.1     
[45] scales_1.4.0       ps_1.9.1           promises_1.5.0     htmltools_0.5.9   
[49] dichromat_2.0-0.1  httpuv_1.6.10      labeling_0.4.3     S7_0.2.1          
[53] stringi_1.7.12     crayon_1.5.3      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
